app/assets/javascripts/discourse/routes/application.js.es6

```diff
@@ -134,16 +134,11 @@ var ApplicationRoute = Discourse.Route.extend({
     editCategory: function(category) {
       var router = this;
 
-      if (category.get('isUncategorizedCategory')) {
-        Discourse.Route.showModal(router, 'editCategory', category);
+      Discourse.Category.reloadById(category.get('id')).then(function (c) {
+        Discourse.Site.current().updateCategory(c);
```

Comment:
While you're in there, instead of `Discourse.Site.current()` please change it to `self.site.updateCategory(c);` where `self` is defined outside of the closure